### ISSUE TITLE
Last log line missing when using multline parsing

### ISSUE THREAD

**Issue by llfab** [carina-studio/ULogViewer#35]
This is my log format (that I am free to change):

```
2024-03-13 09:58:57.616 | TargetingScene3DViewModel-TargetingSleeveView-Sleeve | TID[     1] | W:32/32763/32767 C:1/1000/1000 | Debug     | UpdatePlateScrewPlacements(): called.
2024-03-13 09:58:57.617 | TargetingControlViewModel | TID[     1] | W:32/32763/32767 C:1/1000/1000 | Debug     | OnTargetingModelPlateScrewPlacementAllowedChanged(): called. IsAllowed[True]
2024-03-13 09:59:05.138 | TargetingControlViewModel | TID[     1] | W:32/32764/32767 C:1/1000/1000 | Info      | SetPlateScrew(): PlateHole[Shaft04] called.
2024-03-13 09:59:05.139 | TargetingModel       | TID[     1] | W:32/32764/32767 C:1/1000/1000 | Debug     | SetPlateScrewPlacement(): PlateScrewPlacement[PlateHole[Shaft04] IsSet[True] Position[
  IsValid[True] Head[(124.8769, 6.0189, 38.8442)] Direction[(-0.0602, 0.1358, -0.9889)]
] ScrewType[NonLocking60]]
2024-03-13 09:59:05.140 | TargetingScene3DViewModel-TargetingSleeveView-Sleeve | TID[     1] | W:32/32764/32767 C:1/1000/1000 | Debug     | OnTargetingModelPlateScrewPlacementsChanged(): called. PlateHole[Shaft04]
2024-03-13 09:59:05.141 | TargetingScene3DViewModel-TargetingSleeveView-Sleeve | TID[     1] | W:32/32764/32767 C:1/1000/1000 | Debug     | UpdatePlateScrewPlacements(): called.
2024-03-13 09:59:05.168 | TargetingScene3DViewModel-TargetingSideView-Side | TID[     1] | W:32/32763/32767 C:1/1000/1000 | Debug     | OnTargetingModelPlateScrewPlacementsChanged(): called. PlateHole[Shaft04]
2024-03-13 09:59:22.364 | CaseManager          | TID[     1] | W:32/32765/32767 C:1/1000/1000 | Info      | Exit(): called.
2024-03-13 09:59:22.364 | CaseModel            | TID[     1] | W:32/32765/32767 C:1/1000/1000 | Info      | Exit(): successful.
2024-03-13 09:59:22.365 | PresentationModel    | TID[     1] | W:32/32765/32767 C:1/1000/1000 | Info      | Exit(): successful.
2024-03-13 09:59:22.366 | ImageFileMemoryManager | TID[     1] | W:32/32765/32767 C:1/1000/1000 | Info      | DeleteFileData(): Existing Directory[Temp/MemoryImages] deleted.
2024-03-13 09:59:22.367 | NavappsLogging       | TID[     1] | W:32/32766/32767 C:1/1000/1000 | Info      | Exit():: successful.
2024-03-13 09:59:22.367 | LoggingManager       | TID[     1] | W:32/32766/32767 C:1/1000/1000 | Info      | Exit(): successful.
2024-03-13 09:59:22.367 | PixrNpcApplication   | TID[     1] | W:32/32766/32767 C:1/1000/1000 | Info      | Exit(): successful.
2024-03-13 09:59:22.367 | PixrNpcSystem        | TID[     1] | W:32/32766/32767 C:1/1000/1000 | Info      | Exit(): successful.
2024-03-13 09:59:22.444 | App                  | TID[     1] | W:32/32766/32767 C:1/1000/1000 | Info      | OnDesktopLifetimeExit(): done.
2024-03-13 09:59:22.445 | PixrNpcSystem        | TID[     1] | W:32/32766/32767 C:1/1000/1000 | Info      | Shutdown(): successful.
```

We use multi-line in some areas, hence, I configured the parser like this: 
![image](https://github.com/carina-studio/ULogViewer/assets/31031996/951dcdb6-9fe4-4b76-8be8-e85c07491396)

```
  "LogPatterns": [
    {
      "Regex": "^(?\u003CTimestamp\u003E[\\d\\-]\u002B\\s[\\d\\:\\.]\u002B)\\s\u002B\\|\\s\u002B(?\u003CSourceName\u003E\\S\u002B)\\s\u002B\\|\\s\u002BTID\\[(?\u003CThreadId\u003E[\\d\\s]\u002B)\\]\\s\u002B\\|\\s\u002B(?\u003CWT\u003E\\S\u002B)\\s(?\u003CCT\u003E\\S\u002B)\\s\u002B\\|\\s\u002B(?\u003CLevel\u003E\\S\u002B)\\s\u002B\\|\\s\u002B(?\u003CMessage\u003E.*)",
      "IgnoreCase": true
    },
    {
      "Regex": "^(?\u003CMessage\u003E(?!\\d{4}-\\d{2}-\\d{2}).*)",
      "IgnoreCase": true,
      "IsRepeatable": true,
      "IsSkippable": true
    }
  ],
```

It almost works but it would always miss the last log line. Only if I remove the 2nd regex the last log line would show up. Any solution to this?

![image](https://github.com/carina-studio/ULogViewer/assets/31031996/c1b408c2-dac2-4557-86ac-5d9c70eb1e38)

Note that this log line is missing:
`2024-03-13 09:59:22.445 | PixrNpcSystem        | TID[     1] | W:32/32766/32767 C:1/1000/1000 | Info      | Shutdown(): successful.`

**Comment by llfab** [comment]
Great tool by the way. Just sponsored it!

**Comment by hamster620** [comment]
@llfab thanks for finding the issue, we will fix in next release.

**Comment by hamster620** [comment]
Fixed in [4.0.9.331](https://github.com/carina-studio/ULogViewer/releases/tag/4.0.9.331).
